### ISSUE TITLE
texture_top offset start point wrong

### ISSUE THREAD

**Issue by matthiasmendler** [sirjuddington/SLADE#861]
**SLADE Version**: (3.1.2.4)
**Operating System**: (Linux)

**Issue Details**:
EDIT: I made some mistakes in the first report. Fixed now.

I don' t know the right terminology, but I think these screenshots explain the problem

SW1GRAY, the dark switch, is texture_mid
SW2GRAY, the bright switch is texture_top

texture_mid from scaley=1.00 in 0.10 increasements until scaley=1.50.
In Slade:
![auswahl_037](https://user-images.githubusercontent.com/13328569/38466856-9b4743b2-3b2f-11e8-938f-b4e1908bdeda.png)
In GZDOOM:
![auswahl_039](https://user-images.githubusercontent.com/13328569/38466857-9def24e0-3b2f-11e8-9f93-e5827baea1e7.png)
 Everything is fine. The texture begins in the top and gets scaled from there.

as texture_top  from scaley=1.00 in 0.10 increasements until scaley=1.50.
In Slade:
![grafik](https://user-images.githubusercontent.com/13328569/38466866-c8ef9c7e-3b2f-11e8-9898-764758ee2528.png)
In GZDOOM:
![grafik](https://user-images.githubusercontent.com/13328569/38466871-d2387ea4-3b2f-11e8-9a39-6d95ca3a15a1.png)

Slade scales the texture from the middle of the height, while gzdoom does it from the bottom.

**Comment by matthiasmendler** [comment]
Here is the wad:
[testmap2.zip](https://github.com/sirjuddington/SLADE/files/1887370/testmap2.zip)

**Comment by matthiasmendler** [comment]
So. I had a really really bad solution here before. Now it's just a bad solution, I guess. 
Please don't kill me ;) My c++ skills are quite rudimentary, I never did anything like this before, never used opengl, and I also only used github to get source so far. I tried to get this onto github, but failed, so here as a patch:

```
--- a/src/MapEditor/Renderer/MapRenderer3D.cpp
+++ b/src/MapEditor/Renderer/MapRenderer3D.cpp
@@ -1519,6 +1519,13 @@ void MapRenderer3D::updateLine(unsigned index)
 		quad.colour = colour1;
 		quad.fogcolour = fogcolour1;
 		quad.light = light1;
+
+		// Hack to show upper unpegged Textures with the right offset, as gzdoom handles it
+		if (!upeg)
+		{
+			yoff -= (quad.texture->getHeight())*(1 - sy);
+		}
+		
 		quad.texture = MapEditor::textureManager().getTexture(line->s1()->getTexUpper(), mixed);
 		setupQuadTexCoords(&quad, length, xoff, yoff, ceiling1, ceiling2, !upeg, sx, sy);
 		// Sky hack only applies if both sectors have a sky ceiling
@@ -1691,6 +1698,13 @@ void MapRenderer3D::updateLine(unsigned index)
 		quad.fogcolour = fogcolour2;
 		quad.light = light2;
 		quad.texture = MapEditor::textureManager().getTexture(line->s2()->getTexUpper(), mixed);
+		
+		// Hack to show upper unpegged Textures with the right offset, as gzdoom handles it
+		if (!upeg)
+		{
+			yoff -= (quad.texture->getHeight())*(1 - sy);
+		}
+		
 		setupQuadTexCoords(&quad, length, xoff, yoff, ceiling2, ceiling1, !upeg, sx, sy);
 		if (S_CMPNOCASE(sky_flat, line->frontSector()->getCeilingTex())) quad.flags |= SKY;
 		quad.flags |= BACK;

```

**Comment by matthiasmendler** [comment]
As I said I am new to committing to github, too. Forked this project for two lines of code.
https://github.com/matthiasmendler/SLADE/commit/017b5073638f93f56f0d11dfe9d12fba49702b02

**Comment by sirjuddington** [comment]
It should be fine to submit pull requests, worst that can happen is I'll request changes :P

**Comment by sirjuddington** [comment]
Looks like this was fixed in #963
